R/add_totals.R

```diff
@@ -1,9 +1,67 @@
+#' @title Append a totals row and/or column to a data.frame.
+#'
+#' @description
+#' This function excludes the first column of the input data.frame, assuming it's a descriptive variable not to be summed.  It also excludes other non-numeric columns.
+#'
+#' @param dat an input data.frame with at least one numeric column.
+#' @param which one of "row", "col", or \code{c("row", "col")} 
+#' @param fill if there are multiple non-numeric columns, what string should fill the bottom row of those columns?
+#' @param na.rm should missing values (including NaN) be omitted from the calculations?
+#' @return Returns a data.frame augmented with a totals row, column, or both.
+#' @export
+#' @examples
+#' library(dplyr) # for the %>% pipe
+#' mtcars %>%
+#'   crosstab(am, cyl) %>%
+#'   add_totals()
+
+
+add_totals <- function(dat, which = c("row", "col"), fill = "-", na.rm = TRUE){
```

Comment:
I think there's something to this, and I agree that `adorn_crosstab` will become the outlier (a) because the nomenclature of "crosstab" may go away and (b) if we can fully replicate its functionality with modular pieces like this.  In that case, I do like the common prefix of `adorn_` and could see this as `adorn_totals`.

I want to merge this in now rather than wait for us to figure out #101, since this closes existing bugs.  I'm not sure if it should be `add_totals` when we might later re-route people to `adorn_totals`... maybe `adorn_totals` now.  I think this won't be set in stone until we resolve #101, the big question, and so no matter what we call it now, the name and behavior of the totals function could ultimately change.  Say, if the totals function needs to add an attribute to a tabyl to convey info to `ns_to_percents` down the line.